tools/build_utils.py

```diff
@@ -487,17 +487,43 @@ def install_ngraph_tf(tf_version, venv_dir, ngtf_pip_whl):
     print(ngraph_bridge.__version__)
 
 
+# Note: version can be a branch-name or tag or commit-id
 def download_repo(target_name, repo, version, submodule_update=False):
-    # First download to a temp folder
-    call(["git", "clone", repo, target_name])
+    if not os.path.exists(target_name):
+        call(["git", "clone", repo, target_name])
 
     pwd = os.getcwd()
-    os.chdir(target_name)
+    try:
+        os.chdir(target_name)
+    except:
+        raise Exception("Cannot find dir: " + target_name)
+
+    curr_remote = subprocess.check_output(
+        "git remote get-url origin", shell=True).rstrip()
+    if curr_remote != repo:
```

Comment:
In an existing workspacce if the remote is changed, this checks for that and adjusts